application.go

```diff
@@ -72,132 +72,136 @@ type Application struct {
 	LastTaskFailure       *LastTaskFailure    `json:"lastTaskFailure,omitempty"`
 }
 
+// ApplicationVersions ... a collection of application versions for a specific app in marathin
 type ApplicationVersions struct {
 	Versions []string `json:"versions"`
 }
 
+// ApplicationVersion ... the application version response from marathon
 type ApplicationVersion struct {
 	Version string `json:"version"`
 }
 
+// VersionInfo ... the application versioning details from marathon
 type VersionInfo struct {
 	LastScalingAt      string `json:"lastScalingAt,omitempty"`
 	LastConfigChangeAt string `json:"lastConfigChangeAt,omitempty"`
 }
 
+// NewDockerApplication ... create a default docker application
 func NewDockerApplication() *Application {
 	application := new(Application)
 	application.Container = NewDockerContainer()
 	return application
 }
 
-// The name of the application i.e. the identifier for this application
-func (application *Application) Name(id string) *Application {
-	application.ID = validateID(id)
-	return application
+// Name ... set the name of the application i.e. the identifier for this application
+func (r *Application) Name(id string) *Application {
```

Comment:
Again just a minor thing: usually in this case the first letter is used for self reference. You can look at the glog example as well: https://github.com/golang/glog/blob/master/glog.go#L222
In this case it could be `func (a *Application) Name(id string)`.